dpp/src/identity/state_transition/identity_credit_withdrawal_transition/mod.rs

```diff
@@ -0,0 +1,140 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+use crate::{
+    identity::KeyID,
+    prelude::Identifier,
+    state_transition::{
+        StateTransitionConvert, StateTransitionIdentitySigned, StateTransitionLike,
+        StateTransitionType,
+    },
+    util::json_value::{JsonValueExt, ReplaceWith},
+    ProtocolError,
+};
+
+use super::properties::{
+    PROPERTY_IDENTITY_ID, PROPERTY_OUTPUT, PROPERTY_OWNER_ID, PROPERTY_SIGNATURE,
+    PROPERTY_SIGNATURE_PUBLIC_KEY_ID,
+};
+
+pub mod apply_identity_credit_withdrawal_transition_factory;
+pub mod validation;
+
+pub const IDENTIFIER_FIELDS: [&str; 2] = [PROPERTY_IDENTITY_ID, PROPERTY_OWNER_ID];
+
+#[repr(u8)]
+#[derive(Serialize_repr, Deserialize_repr, PartialEq, Eq, Clone, Copy, Debug)]
+pub enum Pooling {
+    Never = 0,
+    IfAvailable = 1,
+    Standard = 2,
+}
+
+impl std::default::Default for Pooling {
+    fn default() -> Self {
+        Pooling::Standard
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IdentityCreditWithdrawalTransition {
+    pub protocol_version: u32,
+    #[serde(rename = "type")]
+    pub transition_type: StateTransitionType,
+    pub identity_id: Identifier,
+    pub amount: u64,
+    pub core_fee: u64,
+    pub pooling: Pooling,
+    pub output: Vec<u8>,
+    pub signature_public_key_id: KeyID,
+    pub signature: Vec<u8>,
+}
+
+impl std::default::Default for IdentityCreditWithdrawalTransition {
+    fn default() -> Self {
+        IdentityCreditWithdrawalTransition {
+            protocol_version: Default::default(),
+            transition_type: StateTransitionType::IdentityCreditWithdrawal,
+            identity_id: Default::default(),
```

Comment:
You can set only type and `..Default::default()` for the rest

Reply:
This function _is_ `Default::default()`, adding `..Default::default()` will lead to an infinite recursion

Reply:
`{sometheing: value, ..Default::default}` is perfectly fine, just not inside a function called `default`

Reply:
Oh, I see. Thank you.